### ISSUE TITLE
Create "clone agent" function

### ISSUE THREAD

**Issue by cleberjamaral** [jacamo-lang/jacamo-web#39]
In case of an agent which is sharing the aslfile with others, it is necessary a function to allow to edit a unique agent, in this case, "cloning" the source file and pointing the agent that is being edited to this new file.

**Comment by jomifred** [comment]
what about decoupling agents and "source" files? an asl file is a set of plans, not an agent. 
an agent can be built on top of many asl files... 

maybe, we can conceive better the notion of persistence, where all (or part) the state of the agent are stored.

**Comment by cleberjamaral** [comment]
Indeed, we have to develop a way to create "self-contained instances" which were created from models, but after that, in a way, they are independent and may persist. It brings some challenges regarding updates, it is common to expect that many instances can be updated in a simple way after updating their model, we have to think about how it could work.
